pkg/controller/activegate/activegate_update.go

```diff
@@ -66,7 +85,7 @@ func (r *ReconcileActiveGate) findOutdatedPods(
 				logger.Error(err, err.Error())
 			}
 
-			isLatest, err := isLatestFn(logger, instance.Spec.Image, status.ImageID, imagePullSecret)
+			isLatest, err := r.updateService.IsLatest(logger, instance.Spec.Image, status.ImageID, imagePullSecret)
```

Comment:
```suggestion
			isLatest, err := updateService.IsLatest(logger, instance.Spec.Image, status.ImageID, imagePullSecret)
```

Reply:
The reconciler could have a different service than the calling one